responses/__init__.py

```diff
@@ -696,22 +701,27 @@ def __exit__(self, type, value, traceback):
     def activate(self, func):
         return get_wrapped(func, self)
 
+    def _matching_sequence(self):
+        index = 0
+        for response in self.registry:
+            if response.call_count == 0:
+                yield index, response
+            index += 1
+        for response in reversed(self.registry):
+            index -= 1
+            if response.call_count > 0:
+                yield index, response
```

Comment:
Do we need to `yield` the index? It looks unused in all the places this method is called.

Reply:
I needed to add it for the `replace` method (see second commit), which replaces the next match in matching order. Therefore `_matches.index` (now giving the next index in insertion order) needed to be replaced and I thought yielding index together with the match would facilitate the localization without additional helper functions.

Reply:
Ah I missed that as the usage was outside of the modified lines.